src/marinade-mint/marinade-mint.ts

```diff
@@ -1,22 +1,21 @@
-import { Provider, web3, BN } from '@coral-xyz/anchor'
+import { web3, BN } from '@coral-xyz/anchor'
 import { getMint, Mint } from '@solana/spl-token-3.x'
 import { tokenBalanceToNumber } from '../util/conversion'
 
 export class MarinadeMint {
   private constructor(
-    private readonly anchorProvider: Provider,
+    private readonly connection: web3.Connection,
```

Comment:
Appreciate a lot the fact that we're getting rid of anchorProvider, but we should also import only connection from solana/web3js instead of the entire web3 from anchor package